Optimize the custom doc for shaders.

magian1127 opened the pull request:
﻿1. Remove `add_doc` for gdshader files without documentation.
2. Standardize the `class_doc.name` for custom documentation.

Delsin-Yu commented on scene/resources/material.cpp:
```diff
@@ -351,6 +351,9 @@ void ShaderMaterial::_get_property_list(List<PropertyInfo> *p_list) const {
 
 			PropertyInfo info = E->get();
 			info.name = "shader_parameter/" + info.name;
+#if defined(TOOLS_ENABLED) && defined(MODULE_REGEX_ENABLED)
+			info.class_name = info.class_name;
+#endif
```
This doesn't look correct.

anvilfolk commented on scene/resources/shader.cpp:
```diff
@@ -130,10 +130,16 @@ void Shader::get_shader_uniform_list(List<PropertyInfo> *p_params, bool p_get_gr
 	DocData::ClassDoc class_doc;
-	class_doc.name = get_path();
-	class_doc.is_script_doc = true;
+	bool generate_doc = EditorHelp::get_doc_data() != nullptr && Engine::get_singleton()->is_editor_hint() && !get_path().is_empty();
```
This can be simplified if #95821 is merged. You should not need to check whether `get_doc_data() != nullptr`, the helper methods `EditorHelp::add_doc()/remove_doc()` etc can make those checks :)

magian1127 commented:
Modified the code according to #95821

KoBeWi commented on scene/resources/shader.cpp:
```diff
@@ -154,10 +154,15 @@ void Shader::get_shader_uniform_list(List<PropertyInfo> *p_params, bool p_get_gr
 	RenderingServer::get_singleton()->get_shader_parameter_list(shader_rid, &local);
 
-#ifdef TOOLS_ENABLED
+#if defined(TOOLS_ENABLED) && defined(MODULE_REGEX_ENABLED)
```
Is this some rebase leftover? TOOLS_ENABLED implies regex AFAIK.

KoBeWi reviewed (approved):
The `#if`s need to be reverted, otherwise looks good.

magian1127 commented:
> What does this improve specifically? There are no linked bug reports, proposals, or any explanation

https://github.com/godotengine/godot/pull/95821#issuecomment-2369270073

>EDIT2:

magian1127 committed:
Optimize the custom doc for shaders.

Remove add_doc for gdshader files without documentation.
Standardize the class_doc.name for custom documentation.
